aws-controllers-k8s/dev-tools
# build: update Go version and dependencies

This commit updates the Go version requirements for the project to
`1.21.x` and `1.22.x` as well as updates various dependencies to their
latest compatible versions

Dependency changes:

- Update `go.mod` to use Go 1.21 and set the toolchain to go 1.22.1
- Update `github.com/google/go-github` to v61.0.0
- Update `github.com/go-git/go-git` to v5.11.0
- Update other dependencies to their latest versions
- Regenerate mocks to use new go-git imports

Signed-off-by: Amine Hilaly <hilalyamine@gmail.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
